yadro/watcher.cpp

```diff
@@ -532,3 +632,42 @@ void dbuswatcher::onPowerStateChanged(sdbusplus::message::message& m)
         }
     }
 }
+/**
+ * @brief Set BMC Version
+ *
+ * @param ver - new version
+ */
+void dbuswatcher::setBMCVersion(const std::string& ver)
+{
+    if (0 != ver.compare(versionBMC))
+    {
+        auto len = std::min<size_t>(VERSION_MAX_LEN, ver.size());
+        std::string prev(versionBMC);
```

Comment:
Again, I don't see the reason for using `std::string` here instead of static allocation of `char[VERSION_MAX_LEN + 1]` array. Is it just to avoid calling `memcpy`/`strcpy` ?
Maybe change `versionBMC` and `versionSysFW` (or whatever) to `std::string` then?

Reply:
The reason lies in inter-thread-communications.
Pointer to char array in our case will never changed but `std::string` may reallocate his internal buffer.
`net-snmp` calls required c-style strings and they maybe changed in other thread.